app/helpers/second_factor_helper.rb

```diff
@@ -1,35 +0,0 @@
-module SecondFactorHelper
```

Comment:
@featheredtoast Rails helpers are meant to extract logic used in views and isn't really a good place to contain server side logic. What I did here instead is to extract the logic here into a module `SecondFactorManager` and included the module in the `User` class.